tests/test_transforms/test_decomposition/test_deseasonal_transform.py

```diff
@@ -0,0 +1,216 @@
+
+    classic_df = pd.concat([df_1, df_2], ignore_index=True)
+    df = TSDataset.to_dataset(classic_df)
+    df.loc[[df.index[0], df.index[1]], pd.IndexSlice["segment_1", "target"]] = None
+    return TSDataset(df, freq="D")
+
+
+@pytest.fixture
+def ts_seasonal_nan_tails() -> TSDataset:
+    df_1 = get_one_df(period=7, magnitude=1)
+    df_1["segment"] = "segment_1"
+
+    df_2 = get_one_df(period=7, magnitude=2)
+    df_2["segment"] = "segment_2"
+
+    classic_df = pd.concat([df_1, df_2], ignore_index=True)
+    df = TSDataset.to_dataset(classic_df)
+    df.loc[[df.index[-2], df.index[-1]], pd.IndexSlice["segment_1", "target"]] = None
+    return TSDataset(df, freq="D")
+
+
+@pytest.mark.parametrize("model", ["additive", "multiplicative"])
+@pytest.mark.parametrize("df_name", ["df_seasonal_one_segment", "df_seasonal_starting_with_nans_one_segment"])
+def test_transform_one_segment(df_name, model, request):
+    """Test that transform for one segment removes seasonality."""
+    df = request.getfixturevalue(df_name)
+    transform = _OneSegmentDeseasonalityTransform(in_column="target", period=7, model=model)
+    df_transformed = transform.fit_transform(df)
+    df_expected = df.copy()
+    df_expected.loc[~df_expected["target"].isna(), "target"] = 10
+    np.testing.assert_allclose(df_transformed["target"], df_expected["target"], atol=0.3)
+
+
+@pytest.mark.parametrize("model", ["additive", "multiplicative"])
+@pytest.mark.parametrize("ts_name", ["ts_seasonal", "ts_seasonal_starting_with_nans", "ts_seasonal_nan_tails"])
+def test_transform_multi_segments(ts_name, model, request):
+    """Test that transform for all segments removes seasonality."""
+    ts = request.getfixturevalue(ts_name)
+    df_expected = ts.to_pandas(flatten=True)
+    df_expected.loc[~df_expected["target"].isna(), "target"] = 10
+    transform = DeseasonalityTransform(in_column="target", period=7, model=model)
+    transform.fit_transform(ts=ts)
+    df_transformed = ts.to_pandas(flatten=True)
+    np.testing.assert_allclose(df_transformed["target"], df_expected["target"], atol=0.3)
+
+
+@pytest.mark.parametrize("model", ["additive", "multiplicative"])
+@pytest.mark.parametrize("df_name", ["df_seasonal_one_segment", "df_seasonal_starting_with_nans_one_segment"])
+def test_inverse_transform_one_segment(df_name, model, request):
+    """Test that transform + inverse_transform don't change dataframe."""
+    df = request.getfixturevalue(df_name)
+    transform = _OneSegmentDeseasonalityTransform(in_column="target", period=7, model=model)
+    df_transformed = transform.fit_transform(df)
+    df_inverse_transformed = transform.inverse_transform(df_transformed)
+    pd.util.testing.assert_frame_equal(df_inverse_transformed, df)
+
+
+@pytest.mark.parametrize("model", ["additive", "multiplicative"])
+@pytest.mark.parametrize("ts_name", ["ts_seasonal", "ts_seasonal_starting_with_nans"])
+def test_inverse_transform_multi_segments(ts_name, model, request):
```

Comment:
Doesn't we want to add `ts_seasonal_nan_tails`?

Reply:
Yes, you are right!